### PR TITLE
feat: added release modal

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- Modal to release the payment.
- Cancel closes the modal and does nothing. You can also click out to close the modal.
- Clicking on "Release payment" will release the payment, start claim delays
    - Permissions will immediately release the funds and start claim delays.
- Decision methods
    - Options
        - Permissions
            - Visible if the user holds the Payer permissions or Arbitration custom permission in the team.

## Testing

- Create expenditure
- Lock and fund it
- In action sidebar open modal by clicking on temp button
image

Resolves [https://github.com/JoinColony/colonyCDapp/issues/2096](https://github.com/JoinColony/colonyCDapp/issues/2096)

### CODE REVIEW

File: src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/ReleasePaymentModal/consts.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  const userPermissions = getAllUserRoles(colony, user.walletAddress);
+  const userRole = getRole(userPermissions);
+
+  const isPermissionsEnabled =
+    userRole.role === UserRole.Owner ||
+    userRole.role === UserRole.Admin ||
+    userRole.role === UserRole.Payer ||
+    userPermissions.includes(ColonyRole.Arbitration);
```
Comment (jakubcolony) [inline src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/ReleasePaymentModal/consts.ts:27]: It's sufficient just to check if the user has `Arbitration` permission but it must take into account the domain of the expenditure. You can replace it with a single check using `addressHasRoles`:
```ts
addressHasRoles({
  address: user.walletAddress,
  colony,
  requiredRoles: [ColonyRole.Arbitration],
  requiredRolesDomains: [expenditure.nativeDomainId] // will need to pass expenditure as prop
})
```

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Great work @CzarekDryl, I was able to finalize once I locked and funded the expenditure using testing UI. I left a comment regarding permissions validation, otherwise LGTM.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl nice start on this.

1. The modal is missing the icon and styling to match Figma.

image

2. The input and dropdown have the same styling issues from #2085 - 3, 4, 5, 6 and 7 
https://github.com/JoinColony/colonyCDapp/pull/2085#pullrequestreview-1957545540

**Review by ArmandoGraterol — COMMENTED** [review]
According to the design on figma, there should be an icon at the top:
Screenshot 2024-03-27 at 15 01 35

Dialog in this PR:
Screenshot 2024-03-27 at 15 02 05

Has this changed?

EDIT: I didn't see Mel review pointing out the same thing

**Comment by CzarekDryl** [comment]
@melyndav Did you check it? Because it was all fixed here in this PR

**Review by bassgeta — APPROVED** [review]
Looking good, very nice and readable, LGTM :ship:

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl thanks for the update on the icon.

1. The main CTA shouldn't be disabled at anytime. It should be left active, but an error state should show if no option is selected as per Figma here. 

https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=22199-239208&mode=design&t=5gViKhxqOZqOTwev-4

2. Button placement on mobile should be consistent and be stickied to the bottom to stay consistent with every other mobile modals.
image

**Comment by CzarekDryl** [comment]
@jakubcolony @ArmandoGraterol @bassgeta I had to modify this modal and add a form, because now there is validation on onSubmit whether an item has been selected in select.

**Review by melyndav — APPROVED** [review]
Thanks for your work on all the requested updates @CzarekDryl - approving this one from my side.

**Review by jakubcolony — APPROVED** [review]

**Pull request merged**